src/EventHub/EventHub/Cmdlets/Clusters/NewAzureEventHubCluster.cs

```diff
@@ -30,24 +32,35 @@ public class NewAzureRmEventHubCluster : AzureEventHubsCmdletBase
         [Parameter(Mandatory = true, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name")]
         [ResourceGroupCompleter]
         [ValidateNotNullOrEmpty]
-         public string ResourceGroupName { get; set; }
+        public string ResourceGroupName { get; set; }
 
+
+        [CmdletParameterBreakingChange("Name", ChangeDescription = "Name Parameter is being deprecated from " + ClusterResourceIdParameterSet + " without being replaced. ResourceId's are implicit of resource name.")]
         [Parameter(Mandatory = true, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Cluster Name")]
         [Parameter(Mandatory = true, ParameterSetName = ClusterResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Cluster Name")]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
+
         [Parameter(Mandatory = true, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = "Location of Cluster")]
         [ValidateNotNullOrEmpty]
         public string Location { get; set; }
 
+
         [Parameter(Mandatory = false, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "Cluster Capacity (CU), curerntrly, allowed value = 1")]
         [ValidateNotNullOrEmpty]
         public int? Capacity { get; set; }
 
+
         [Parameter(Mandatory = false, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "Hashtables which represents resource Tags for Clusters")]
         public Hashtable Tag { get; set; }
 
+
+        [Parameter(Mandatory = false, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "Hashtables which represents resource Tags for Clusters")]
+        public SwitchParameter SupportsScaling { get; set; }
+
+
+        [CmdletParameterBreakingChange("ResourceId", ChangeDescription = "ResourceId Parameter is being deprecated from without being replaced.")]
```

Comment:
Why you are intend to deprecate ResourceId. The reason why you are deprecating Name from ClusterResourceIdParameterSet is we have ResourceId. Do you mean deprecated from ClusterPropertiesParameterSet?

Reply:
We want to restrict the usage of -ResourceId to only Set-Az, Get-Az and Remove-Az cmdlets.

Reply:
We want to keep Name in ClusterPropertiesParameterSet and completely Remove ClusterResourceIdParameterSet  from New-Az cmdlets.

Reply:
It will be a breaking change. It will not approve in regular release